src/database/sqlServer/dbService.ts

```diff
@@ -147,6 +147,33 @@ export function deleteUnits(units: Unit[]): Promise<boolean> {
     });
 }
 
+export function createDenormalized(name: string, unitType: UnitType): Promise<number> {
+  const request = new sql.Request(connectionPool);
+  return request.query(`
+    insert into Denormalized (name , unitType)
+    output Inserted.id
+    values ('${name}', '${unitType}');`)
+    .then((res) => res.recordset[0].id)
+    .catch((error) => {
+      throw error;
+    });
+}
+
+export function createUnit(divisionId: number| undefined, brigadeId: number | undefined,
+  battalionId: number | undefined, companyId: number | undefined): Promise<boolean> {
+  const request = new sql.Request(connectionPool);
+  const query = `
+  insert into Unit (companyID , battalionID, brigadeID, divisionID)
+  values (${companyId || null}, ${battalionId || null}
+  , ${brigadeId || null}, ${divisionId} );`;
```

Comment:
Instead of using | undefined and companyId || null, will optional parameter looks better?